### PR TITLE
DEVPROD-1976: Migrate patch tasks table to LeafyGreen table

### PR DESCRIPTION
**Pull request by minnakt**
DEVPROD-1976

### Description
Updates tasks table to use LeafyGreen table. TBH `disableAnimations` doesn't really help that much, it almost seems the exact same.

### Screenshots
Screenshot 2024-03-21 at 6 36 16 PM

### Testing
- On beta against MongoDB project
- Updated a lot of Cypress tests

### CODE REVIEW

**Review by khelif96 — COMMENTED** [review]
Looks great I think you may have also fixed the bug from https://jira.mongodb.org/browse/DEVPROD-5751

File: cypress/support/commands.ts [hunk 1]
```diff
@@ -68,6 +68,7 @@ Cypress.Commands.add("logout", () => {
 });
 
 /* toggleTableFilter */
+// TODO: Delete this command.
```
[the remark this answers is not in the document]
Comment (minnakt) [reply cypress/support/commands.ts]: just deleted it after merging in the downstream tasks pr!

File: src/components/Table/BaseTable.tsx [hunk 2]
```diff
@@ -226,10 +237,6 @@ const RenderableRow = <T extends LGRowData>({
   </Row>
 );
 
-const StyledTable = styled(Table)`
-  transition: none !important;
```
[the remark this answers is not in the document]
Comment (minnakt) [reply src/components/Table/BaseTable.tsx:230]: Discussed on Slack, we found out it was fixed by https://github.com/mongodb/leafygreen-ui/commit/bb8987d60d5ee2d801629bf72e5ef3ebcb382ed9

**Review by khelif96 — APPROVED** [review]
Great job! The collapse functionality is pretty unresponsive when there are 100 rows but I think its more likely a problem with the LG table

**Comment by minnakt** [comment]
Will re-open in UI repo when LeafyGreen provides a fix

**Pull request closed without merging**